### DISCUSSION TITLE
`next/script`'s `beforeInteractive` strategy should not be used outside of `pages/_document.js`.

### DISCUSSION THREAD

**Discussion by Pnlvfx** [vercel/next.js#46508] [Help]
Why is this error still active in next 13.2.2? As I've read in the beta docs I can use next/script in the root layout, ot not?

**Comment by ciruz** [comment]
That's right, `beforeInteractive` needs to be placed inside the root layout. Do you have some code?

**Comment by Pnlvfx** [comment]
I'm using a root layout and then another layout where I use my header `example: (header)/layout`, that is not present in all pages, and I put the script inside here, and see this eslint error. But even if I use it in the rootLayout I see the same error.

const Layout = ({ children }: ChildrenProps) => {
  return (
    <>
      {process.env.NODE_ENV === 'production' && (
          <Script
            async
            strategy="beforeInteractive"
            src="https://pagead2.googlesyndication.com/pagead/js/adsbygoogle.js?client=ca-pub-**"
            crossOrigin="anonymous"
          />
      )}
    </>
  )
}

export default Layout

**Reply by ciruz** [reply]
The beta docs say it's okay, there is even an example with ``beforeInteractive`` in the new ``rootLayout`` https://beta.nextjs.org/docs/api-reference/components/script#beforeinteractive

Sounds like a bug to me, it's weird, it works for me on TypeScript but not on JavaScript.

I made an issue for this https://github.com/vercel/next.js/issues/46549
